File: tests/abi/wayland_frame_layout.cpp
#include "pluginterfaces/gui/iwaylandframe.h"

#include <cstdio>

template <typename T>
static void print_iid (const char* name, const T& tuid)
{
	std::printf ("%s iid", name);
	for (int index = 0; index < 16; ++index)
		std::printf (" %02X", static_cast<unsigned char> (tuid[index]));
	std::printf ("\n");
}

int main ()
{
	print_iid ("IWaylandHost", Steinberg::IWaylandHost_iid);
	print_iid ("IWaylandFrame", Steinberg::IWaylandFrame_iid);
	return 0;
}

File: tools/wayland_frame_layout.zig
const std = @import("std");
const wayland = @import("vst3-zig").pluginterfaces.gui.iwaylandframe;

pub fn main() !void {
    const stdout = std.io.getStdOut().writer();
    try printTuid(stdout, "IWaylandHost", wayland.iwayland_host_iid);
    try printTuid(stdout, "IWaylandFrame", wayland.iwayland_frame_iid);
}

fn printTuid(writer: anytype, comptime name: []const u8, bytes: [16]u8) !void {
    try writer.print("{s} iid", .{name});
    for (bytes) |byte| {
        try writer.print(" {X:0>2}", .{byte});
    }
    try writer.writeByte('\n');
}
